Testing/UnitTests/xASL_ut_function_xASL_im_ResampleLinearFair.m

```diff
@@ -0,0 +1,177 @@
+function UnitTest = xASL_ut_function_xASL_im_ResampleLinearFair(TestRepository)
```

Comment:
Here we should test that when we make the voxel-size 3 times as small, the sum, mean, median, min, max, etc remain identical to the original voxel-size. Then we make the voxel-size 3 times as large again, and do the same test. Even negligible or small deviations should not be allowed, as that was the whole purpose of the fair linear interpolation.

Reply:
Let's keep my function as subfunction in the unit testing, because this is what we should test, that it keeps the regridding "fair" by giving the identical result as my "more manual" function did.

Reply:
Option to use the previous method: 7f3d4c5

Reply:
Thanks, though I wouldn't call it "old method". The methods are not newer or older, the difference is the matrix operation or the iteration over lines/voxels. Probably better to call them fast vs slow.

Reply:
Yes, though if Jan's method actually works in all cases, then we could keep my function in the unit test only.
Though I wonder if Jan's method is Matlab-version dependent. The unit test with the old function would find this.

Reply:
It does some tri-linear resampling and because of that there are some super minimal differences. I wouldn't make the unit test even more complex. I personally like it the way it is right now.

Reply:
Super simple function - no reason this will differ across Matlab versions. I would keep the new version only. Especially since we only use it in one case of resampling a colormap.

Reply:
> Super simple function - no reason this will differ across Matlab versions. I would keep the new version only. Especially since we only use it in one case of resampling a colormap.

Agree 👍